blockchain/ethereum/internal/chain.go

```diff
@@ -61,18 +68,21 @@ func (cb *ChainBackend) ValidateContracts(adjAddr, assetAddr pwallet.Address) er
 }
 
 // DeployAdjudicator deploys the adjudicator contract.
-func (cb *ChainBackend) DeployAdjudicator() (pwallet.Address, error) {
+func (cb *ChainBackend) DeployAdjudicator(onChainAddr pwallet.Address) (pwallet.Address, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), cb.TxTimeout)
 	defer cancel()
-	addr, err := pethchannel.DeployAdjudicator(ctx, *cb.Cb)
+
+	acc := accounts.Account{Address: pethwallet.AsEthAddr(onChainAddr)}
+	addr, err := pethchannel.DeployAdjudicator(ctx, *cb.Cb, acc)
 	return pethwallet.AsWalletAddr(addr), errors.Wrap(err, "deploying adjudicator contract")
 }
 
 // DeployAsset deploys the asset holder contract, setting the adjudicator address to given value.
-func (cb *ChainBackend) DeployAsset(adjAddr pwallet.Address) (pwallet.Address, error) {
+func (cb *ChainBackend) DeployAsset(adjAddr, cred pwallet.Address) (pwallet.Address, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), cb.TxTimeout)
 	defer cancel()
 
-	addr, err := pethchannel.DeployETHAssetholder(ctx, *cb.Cb, pethwallet.AsEthAddr(adjAddr))
+	acc := accounts.Account{Address: pethwallet.AsEthAddr(cred)}
+	addr, err := pethchannel.DeployETHAssetholder(ctx, *cb.Cb, pethwallet.AsEthAddr(adjAddr), acc)
```

Comment:
Is this onChainAcc or receiverAcc ?

Reply:
It is onChainAcc, [renamed](https://github.com/hyperledger-labs/perun-node/pull/114/commits/aebcf94494c5d13ad272ecadd28edfe7b22dc659) the variable.